src/test/java/com/codeborne/selenide/collections/ItemWithTextTest.java

```diff
@@ -0,0 +1,61 @@
+package com.codeborne.selenide.collections;
+
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.ex.TextsMismatch;
+import com.codeborne.selenide.impl.WebElementsCollection;
+import org.assertj.core.api.WithAssertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static com.codeborne.selenide.Mocks.mockCollection;
+import static com.codeborne.selenide.Mocks.mockElement;
+
+class ItemWithTextTest implements WithAssertions {
+  private SelenideElement element1 = mockElement("Test-One");
+  private SelenideElement element2 = mockElement("Test-Two");
+  private SelenideElement element3 = mockElement("Test-Three");
+  private WebElementsCollection collection = mockCollection("Collection description", element1, element2, element3);
+
+  @Test
+  void applyOnCorrectElementText() {
+    assertThat(new ItemWithText("Test-One")
+      .test(collection.getElements()))
+      .isTrue();
+  }
+
+  @Test
+  void applyOnWrongElementText() {
+    assertThat(new ItemWithText("Test-X")
+      .test(collection.getElements()))
+      .isFalse();
+  }
+
+  @Test
+  void testToString() {
+    assertThat(new ItemWithText("Test-One"))
+      .hasToString("Text Test-One");
+  }
+
+  @Test
+  void testApplyWithEmptyList() {
+    WebElementsCollection emptyCollection = mockCollection("empty collection");
+    assertThat(new ItemWithText("Test-X")
+      .test(emptyCollection.getElements()))
+      .isFalse();
+  }
+
+  @Test
+  void failOnMatcherError() {
+    String expectedText = "Won't exist";
+    assertThatThrownBy(() -> new ItemWithText(expectedText)
+      .fail(collection,
+        collection.getElements(),
+        new Exception("Exception message"), 10000)).isInstanceOf(TextsMismatch.class)
+      .hasMessageContaining(
+        String.format(String.format("Texts mismatch" +
```

Comment:
I think we should have another error message instead of "Texts mismatch". Something like "element with text not found".

Reply:
fixed